modules/src/ics24_host/path.rs

```diff
@@ -41,6 +54,14 @@ pub enum Path {
         channel_id: ChannelId,
         sequence: Sequence,
     },
+    Upgrade(ClientUpgradePath),
+}
+
+/// Paths that are specific for client upgrades.
+#[derive(Clone, Debug)]
```

Comment:
Let's always add `PartialEq, Eq, PartialOrd, Ord, Hash` bounds whenever possible, as equality is often necessary (eg. in tests) and `*Ord` bounds allow storing those types in `BTreeMap` while the `Hash` bound allows storing them in `HashMap`.